dapp_runner/runner/runner.py

```diff
@@ -143,7 +169,7 @@ def dapp_started(self) -> bool:
         clusters have been started and remain running.
         """
 
-        return all(
+        return self._startup_finished and all(
```

Comment:
1. It's unclear to me why `_startup_finished` is necessary here - can we have all instances `running` before `_startup_finished`?
2. If we can, then I think the docstring should include this.

Reply:
it's necessary because earlier, we had assumed `dapp_started` would only be called after `start_services` has finished... now it's called while the services are still starting -> and it may well happen the first call will occur before any clusters have been created, thus, wrongly reporting the dapp as started when it's _just_ began starting

Reply:
OK, now I understand.

I think something like
```
len(self.clusters) == len(self.descriptor.services)
```
could be better (it's usually good to get rid of a single-purpose attribute when we already have the same logic available in some other way), but I don't insist.

Either way, I think the docstring could be clearer (as we don't only care about `commisioned service clusters`, but also about clusters that are yet to be started).

Reply:
I feel more secure with an explicit flag here, so I'd leave it here if you don't insist...

Reply:
I don't :)